core/vm/contracts.go

```diff
@@ -358,3 +374,68 @@ func (c *bn256Pairing) Run(input []byte) ([]byte, error) {
 	}
 	return false32Byte, nil
 }
+
+type blake2F struct{}
+
+func (c *blake2F) RequiredGas(input []byte) uint64 {
+	if len(input) != 213 {
+		// Input is malformed, we can't read the number of rounds.
+		// Precompile can't be executed so we set its price to 0.
+		return 0
+	}
+
+	rounds := binary.BigEndian.Uint32(input[209:213])
+	return uint64(rounds)
+}
+
+var (
+	blake2FInputLength = 213
```

Comment:
this should be a `const`, no?

Reply:
Yeah, it could be:
```
const blake2FInputLength = 213

var errBlake2FIncorrectInputLength = errors.New(
	"input length for Blake2 F precompile should be exactly 213 bytes",
)
```

but thought it's better to initialize them in one block, similarly how it's done above the `bn256Pairing` struct definition. I wasn't sure about the conventions used in the project.

Please let me know what's the preference here and I'll alter it.

Reply:
I think `const` is preferrable, rather than calling it a `var` just for the sake of indentation...

Reply:
Good point, updated.